url933dd5727b_AppleLiang_Testing_Project_tgz-pJ8-org_joda_time_chrono_gj_MainTestJ8/src/org/joda/time/TestDateTimeZoneCutover.java

```diff
@@ -0,0 +1,1340 @@
+
+    //-----------------------------------------------------------------------
+    public void test_MockGazaIsCorrect() {
+        DateTime pre = new DateTime(CUTOVER_GAZA - 1L, MOCK_GAZA);
+        assertEquals("2007-03-31T23:59:59.999+02:00", pre.toString());
+        DateTime at = new DateTime(CUTOVER_GAZA, MOCK_GAZA);
+        assertEquals("2007-04-01T01:00:00.000+03:00", at.toString());
+        DateTime post = new DateTime(CUTOVER_GAZA + 1L, MOCK_GAZA);
+        assertEquals("2007-04-01T01:00:00.001+03:00", post.toString());
+    }
+
+    public void test_getOffsetFromLocal_Gaza() {
+        doTest_getOffsetFromLocal_Gaza(-1, 23, 0, "2007-03-31T23:00:00.000+02:00");
+        doTest_getOffsetFromLocal_Gaza(-1, 23, 30, "2007-03-31T23:30:00.000+02:00");
+        doTest_getOffsetFromLocal_Gaza(0, 0, 0, "2007-04-01T01:00:00.000+03:00");
+        doTest_getOffsetFromLocal_Gaza(0, 0, 30, "2007-04-01T01:30:00.000+03:00");
+        doTest_getOffsetFromLocal_Gaza(0, 1, 0, "2007-04-01T01:00:00.000+03:00");
+        doTest_getOffsetFromLocal_Gaza(0, 1, 30, "2007-04-01T01:30:00.000+03:00");
+        doTest_getOffsetFromLocal_Gaza(0, 2, 0, "2007-04-01T02:00:00.000+03:00");
+        doTest_getOffsetFromLocal_Gaza(0, 3, 0, "2007-04-01T03:00:00.000+03:00");
+        doTest_getOffsetFromLocal_Gaza(0, 4, 0, "2007-04-01T04:00:00.000+03:00");
+        doTest_getOffsetFromLocal_Gaza(0, 5, 0, "2007-04-01T05:00:00.000+03:00");
+        doTest_getOffsetFromLocal_Gaza(0, 6, 0, "2007-04-01T06:00:00.000+03:00");
+    }
+
+    private void doTest_getOffsetFromLocal_Gaza(int days, int hour, int min, String expected) {
+        DateTime dt = new DateTime(2007, 4, 1, hour, min, 0, 0, DateTimeZone.UTC).plusDays(days);
+        int offset = MOCK_GAZA.getOffsetFromLocal(dt.getMillis());
+        DateTime res = new DateTime(dt.getMillis() - offset, MOCK_GAZA);
+        assertEquals(res.toString(), expected, res.toString());
+    }
+
+    public void test_DateTime_roundFloor_Gaza() {
+        DateTime dt = new DateTime(2007, 4, 1, 8, 0, 0, 0, MOCK_GAZA);
+        assertEquals("2007-04-01T08:00:00.000+03:00", dt.toString());
+        DateTime rounded = dt.dayOfMonth().roundFloorCopy();
+        assertEquals("2007-04-01T01:00:00.000+03:00", rounded.toString());
+    }
+
+    public void test_DateTime_roundCeiling_Gaza() {
+        DateTime dt = new DateTime(2007, 3, 31, 20, 0, 0, 0, MOCK_GAZA);
+        assertEquals("2007-03-31T20:00:00.000+02:00", dt.toString());
+        DateTime rounded = dt.dayOfMonth().roundCeilingCopy();
+        assertEquals("2007-04-01T01:00:00.000+03:00", rounded.toString());
+    }
+
+    public void test_DateTime_setHourZero_Gaza() {
+        DateTime dt = new DateTime(2007, 4, 1, 8, 0, 0, 0, MOCK_GAZA);
+        assertEquals("2007-04-01T08:00:00.000+03:00", dt.toString());
+        try {
+            dt.hourOfDay().setCopy(0);
+            fail();
+        } catch (IllegalFieldValueException ex) {
+            // expected
+        }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 106:113, 117:124, 322:329, 333:340, 521:529, 533:541, 852:860, 864:872.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.